core/src/main/java/org/infinispan/expiration/impl/ClusterExpirationManager.java

```diff
@@ -126,6 +117,8 @@ private boolean purgeInMemoryContents(LocalizedCacheTopology topology) {
             start = timeService.time();
          }
          // We limit it so there is only so many non blocking expiration removals done at the same time
+         // Note the addition to the queue shouldn't ever block but rather pollForCompletion when we are waiting for
+         // prior tasks to complete
```

Comment:
Why keep the old comment?

Reply:
It still seems somewhat applicable to me, as we do this to limit the number of inflight expirations.

Reply:
How could limiting the size of the queue limit the number of inflight expirations, if `add()` never blocks?

Reply:
I think we are thinking of limiting in two different ways. You are thinking on the queue, I am thinking on the reaper operation itself. Let me change it to something more descriptive :)

Reply:
The comment is just above the declaration of the queue, why would "it" not refer to the queue?

Reply:
As I mentioned, I was changing it.

Reply:
As I mentioned, I was changing it.